data_collection.ipynb

```diff
@@ -0,0 +1,1194 @@
+       "13  Oct 25, 2022  210.10  224.35  210.00  222.42    222.42   96,507,900\n",
+       "14  Oct 24, 2022  205.82  213.50  198.59  211.25    211.25  100,446,800\n",
+       "15  Oct 21, 2022  206.42  214.66  203.80  214.44    214.44   75,713,800\n",
+       "16  Oct 20, 2022  208.28  215.55  202.00  207.28    207.28  117,798,100\n",
+       "17  Oct 19, 2022  219.80  222.93  217.78  222.04    222.04   66,571,500\n",
+       "18  Oct 18, 2022  229.50  229.82  217.25  220.19    220.19   75,891,900\n",
+       "19  Oct 17, 2022  210.04  221.86  209.45  219.35    219.35   79,428,800\n",
+       "20  Oct 14, 2022  224.01  226.26  204.16  204.99    204.99   94,124,500\n",
+       "21  Oct 13, 2022  208.30  222.99  206.22  221.72    221.72   91,483,000\n",
+       "22  Oct 12, 2022  215.33  219.30  211.51  217.24    217.24   66,860,700\n",
+       "23  Oct 11, 2022  220.95  225.75  215.00  216.50    216.50   77,013,200\n",
+       "24  Oct 10, 2022  223.93  226.99  218.36  222.96    222.96   67,925,000\n",
+       "25  Oct 07, 2022  233.94  234.57  222.02  223.07    223.07   83,916,800\n",
+       "26  Oct 06, 2022  239.44  244.58  235.35  238.13    238.13   69,298,400\n",
+       "27  Oct 05, 2022  245.01  246.67  233.27  240.81    240.81   86,982,700\n",
+       "28  Oct 04, 2022  250.52  257.50  242.01  249.44    249.44  109,578,500\n",
+       "29  Oct 03, 2022  254.50  255.16  241.01  242.40    242.40   98,363,500"
+      ]
+     },
+     "execution_count": 21,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "driver = webdriver.Chrome(executable_path=\"chromedriver.exe\")\n",
+    "url = 'https://finance.yahoo.com/quote/TSLA/history?p=TSLA'\n",
+    "driver.get(url)\n",
+    "\n",
+    "data =[]\n",
+    "for i in range(1,31):\n",
+    "    date = driver.find_element(\"xpath\", '//*[@id=\"Col1-1-HistoricalDataTable-Proxy\"]/section/div[2]/table/tbody/tr['+str(i)+']/td[1]/span').text\n",
+    "    open = driver.find_element(\"xpath\", '//*[@id=\"Col1-1-HistoricalDataTable-Proxy\"]/section/div[2]/table/tbody/tr['+str(i)+']/td[2]/span').text\n",
+    "    high = driver.find_element(\"xpath\", '//*[@id=\"Col1-1-HistoricalDataTable-Proxy\"]/section/div[2]/table/tbody/tr['+str(i)+']/td[3]/span').text\n",
+    "    low = driver.find_element(\"xpath\", '//*[@id=\"Col1-1-HistoricalDataTable-Proxy\"]/section/div[2]/table/tbody/tr['+str(i)+']/td[4]/span').text\n",
+    "    close = driver.find_element(\"xpath\", '//*[@id=\"Col1-1-HistoricalDataTable-Proxy\"]/section/div[2]/table/tbody/tr['+str(i)+']/td[5]/span').text\n",
+    "    adj_close = driver.find_element(\"xpath\", '//*[@id=\"Col1-1-HistoricalDataTable-Proxy\"]/section/div[2]/table/tbody/tr['+str(i)+']/td[6]/span').text\n",
+    "    volume = driver.find_element(\"xpath\", '//*[@id=\"Col1-1-HistoricalDataTable-Proxy\"]/section/div[2]/table/tbody/tr['+str(i)+']/td[7]/span').text\n",
```

Comment:
What other options do we have apart from these variables that can be used for our pre-processing.